ntrack/listener.go

```diff
@@ -0,0 +1,91 @@
+package ntrack
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type trackingListener struct {
+	net.Listener
+	stats *Stats
+}
+
+func NewInstrumentedListener(lis net.Listener) (net.Listener, *Stats) {
+	listenerStats := &Stats{}
+	listenerStats.init()
+
+	return &trackingListener{
+		Listener: lis,
+		stats:    listenerStats,
+	}, listenerStats
+}
+
+func (tl *trackingListener) Accept() (net.Conn, error) {
+	conn, err := tl.Listener.Accept()
+	stats.RecordWithTags(context.TODO(), []tag.Mutator{tag.Upsert(tl.stats.TagSuccess, fmt.Sprintf("%v", err == nil))}, tl.stats.ListenerAccepted.M(1))
+	if err != nil {
+		return nil, errors.Wrap(err, "accept from base listener")
+	}
+
+	atomic.AddInt64(&tl.stats.openConnections, 1)
```

Comment:
Use return value from this statement rather than load on the next line? Same comment for other uses.